dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -353,9 +353,14 @@ private static ChatCompletionsOptions CreateChatCompletionsOptions(ChatRequestSe
             NucleusSamplingFactor = (float?)requestSettings.TopP,
             FrequencyPenalty = (float?)requestSettings.FrequencyPenalty,
             PresencePenalty = (float?)requestSettings.PresencePenalty,
-            ChoiceCount = requestSettings.ResultsPerPrompt
+            ChoiceCount = requestSettings.ResultsPerPrompt,
         };
 
+        if (requestSettings.Functions is not null)
+        {
+            options.Functions = requestSettings.Functions.ToFunctionDefinitions();
```

Comment:
I don't see the FunctionCall option ([here](https://github.com/Azure/azure-sdk-for-net/blob/d8a846377185b5f4d2f30c10746f309ae879e127/sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletionsOptions.cs#L68C61-L68C61))exposed, while I do think that that is important to steer the behaviour, with None, when you want to turn it off, or with a function name when you just want it to use that function.

Reply:
It felt a little redundant -- if a list of functions is provided, FunctionCall is auto by default. If no functions provided, FunctionCall is none by default. However, my solution does not (yet) handle the single function option. I was concerned about taking a direct dependency on the Azure Sdk (via FunctionDefinition). But with Dmytro's suggestion in another comment, if we add a new class model to contain function information, that would make it easier to support that option.

Reply:
I've added the `FunctionCall` parameter to the `OpenAIRequestSettings`. So the behavior would be:
- **to use a specific function:** provide that function via `FunctionCall`, the model would choose that function and populate the parameters
- **for the `auto` option:** provide a list of functions via `Functions`, the model can optionally choose a function from this list
- **for the `none` option:** leave both `Functions` and `FunctionCall` empty/null

Reply:
@eavanvalkenburg I'm looking at your Python PR to see how you handled this. It looks like you set `function_call` to either "auto", "none", or the name of a function. If the latter, you look for that function name in the list of `functions`. Is that right? I can mirror that behavior here by changing `FunctionCall` to be a string instead.

Reply:
that is correct, the name should refer to the name of the function that should be called (where the model mostly supplies the parameters), and then I filter the list of functions so that we save tokens.